### ISSUE TITLE
A/B Tesing on [IzzyOnDroid ](https://apt.izzysoft.de/fdroid/)

### ISSUE THREAD

**Issue by UriahShaulMandel** [UriahShaulMandel/BaldPhone#38]
I can't change package names (and neither can I sign), so I'd need signed APKs with a different package name to start with. You could use a [build variant](https://developer.android.com/studio/build/build-variants) for that, which basically is identical to the "standard one" with the exception of using a different package name (I'd suggest appending `.beta` to the real one, so `com.bald.uriah.baldphone.beta` – Gradle has an automatism for that even, `applicationIdSuffix` (see the example on the linked page), so you'd just need to provide the suffix and it would join the two on build; F-Droid's build server doesn't support that currently, so only use it for the beta build, please). A "product flavor" (described on the same page) would work as well, whichever is easier for you.

Then we need to take care with the tags/releases, to have the F-Droid build process ignore the betas (and mine ignore the "real releases"). Update check at F-Droid must use `%c` currently (tag name corresponds to versionCode) – so for the betas, you could use `%v` instead (tag name corresponds to versionName). Easy to adjust on my end (just let me know when it's ready), and no adjustments needed for the official repo. Note that for version compare, my update checker will ignore letters (so e.g. `1.0.2-alpha3` would be considered newer than `1.0.2-beta1` because `1.0.2-3 > 1.0.2-1` – you have no idea what funny tag names people come up with making that necessary).

_Originally posted by @IzzySoft in https://github.com/UriahShaulMandel/BaldPhone/issues/8#issuecomment-519126231_

**Comment by UriahShaulMandel** [comment]
Hi @IzzySoft ,
I will take care of this in the following weeks

**Comment by IzzySoft** [comment]
Take your time. I switch versions when you're ready. Also feel free to move… ah, wait, you just did that. Good. Just wanted to suggest splitting that out to its own issue :laughing:

**Comment by TPS** [comment]
@UriahShaulMandel When do you expect to start releasing these β builds?

**Comment by UriahShaulMandel** [comment]
Hi @TPS , I'm a high school student and sadly in the next 2 weeks will be on extremely limited availability, because I have multiple important tests coming up real soon. After that should have more time to work on BaldPhone :)
β builds requires a change in the updating mechanism, and will take a few days to implement, days I currently don't have...

There are some things I think are a bit more urgent, such as Video Tutorials in English, Keyboards in different languages and fixing some bugs which prevent some users from using BaldPhone.

With that said, **I acknowledge the importance of β builds, and will implement this mechanism as soon as I can.** I hope that in the next 2 months or so, but sadly can't promise, I'm a student and don't have full control over my time... 


You will be the first one to know when they'll be up :),
Also, you can see here: https://github.com/UriahShaulMandel/BaldPhone/releases that I release update quite frequently, so new features will hopefully be up every week or so.
If you have any suggestion\ferature you want me to implement, feel free to open an issue :)
Uriah Shaul Mandel

**Comment by TPS** [comment]
No stress/pressure meant from my side.…  I was just trying to get a feel for your timescale.  Carry on w/ your studies & good luck!

**Comment by IzzySoft** [comment]
> I will take care of this in the following ~weeks~ months :sweat_smile:

I suggest a correction here:

> I will take care of this in the following ~weeks months~ years

:speak_no_evil:

Ahem, meant to ask, without any pressure: any progress on this?

**Comment by UriahShaulMandel** [comment]
Yeah, sadly, I have absolutely no time to work on BaldPhone, and now will have even less. I only maintain whats critical for it existence.
for 9 months,  I won't be at my home at all, and almost and won't have access to laptop and internet, so it won't get better any time soon,
But in 9 months I may have more time lol
So currently there's no progress on that, but if anything will change I'll notify here:)
And again @IzzySoft thank you so much for your help! You can look at the stats at baldphone.com/stats, its kinda amazing how many people downloaded it!
Uriah

**Comment by IzzySoft** [comment]
Thanks for your fast response! Congrats to the downloads – and, oh well… all the best for getting through those 9 months!

Btw: So ther than your post on Reddit suggests, you were able to get your app listed at Play? Then I might have to remove my comment on that other issue, which you also could close as being solved…

And yes, I always promote your app when I get the chance to (like in my courses) :smiley:

**Comment by TPS** [comment]
@IzzySoft GPlay has this @ https://play.google.com/store/apps/details?id=com.bald.uriah.baldphone.gp.

**Comment by IzzySoft** [comment]
Well, that actually sounds good: We can adopt that package name for the test releases in my repo (and translate the suffix as "general purpose" :rofl:) Signatures should then match again (I hope @UriahShaulMandel didn't hand over his signing key to Google?) – so people could not only cross-update, but also install both versions in parallel (the F-Droid one "for real", the version from my repo "for testing").

What do you think, @UriahShaulMandel? If that sounds good to you, simply attach *both* `*.apk` files to `releases/` after having decided on a naming pattern (`app-release-<flavor>.apk` might be a good idea, so there's be `app-release-fdroid.apk` and `app-release-gplay.apk`; alternatively: `<packageName>.apk` would work as well, resulting in `com.bald.uriah.baldphone.apk` and `com.bald.uriah.baldphone.gp.apk`) so I can tell my updater which one to fetch. The flavor variant has the advantage of being more intuitive for those not that tech-savvy.

Oh, all of course when time permits – no pressure, no need to hurry unless *you* feel like you wish.

**Comment by IzzySoft** [comment]
@UriahShaulMandel is there any news on this? If I'm not mistaken, my repo currently still carries the "default APK". Just wondering if it should stay like this?
